proposals/primer/application.bs

```diff
@@ -0,0 +1,353 @@
+  path: snippets/solidshapes.example/shapes/Project.shex
+  highlight: shex
+  show: 6-100
+  </pre>
+  <figcaption>Project shape (ShEx)</figcaption>
+</figure>
+
+<figure>
+  <pre class=include-code>
+  path: snippets/solidshapes.example/shapes/Task.shex
+  highlight: shex
+  show: 6-100
+  </pre>
+  <figcaption>Task shape (ShEx)</figcaption>
+</figure>
+
+Issue: Reference shape path documentation
+
+# Application # {#appliction}
+
+In this primer we will consider case of an application
+denoted by <code>https://projectron.example/#app</code>
+Every application has public WebID Document providing information about it.
+
+
+
+# Authorization Flow # {#authorization-flow}
+
+User always needs to authenticate first.
+For the rest of this primer we'll be assuming that user has already authenticated,
+and that applicaction know WebID of the authenticated user.
+
+## Authorization Agent Discovery ## {#authorization-agent-discovery}
+
+Every user has an Authorization Agent which can be discovered from
+their WebID Document via <code>interop:hasAuthorizationAgent</code> predicate.
+
+<figure>
+  <pre class=include-code>
+  path: snippets/alice.example/alice.example.ttl
+  highlight: turtle
+  show: 3-100
+  line-start: 1
+  line-highlight: 3
+  </pre>
+  <figcaption>Alice's WebID document</figcaption>
+</figure>
+
+Here we see that Alice designates <code>https://auth.alice.example/</code> as their Authorization Agent.
+
+## Application Registration Discovery ## {#application-registration-discovery}
+
+Application can discover Application Registration created for it by the user,
+by performing <code>HEAD</code> or <code>GET</code> request on IRI denoting
+user's Authorization Agent. Response will include HTTP Link header relating
+Application Registration to the application making the request using
+<code>http://www.w3.org/ns/solid/interop#registeredApplication</code>
+as link relation.
+
+<figure>
+  <pre highlight=http>
+    HEAD / HTTP/1.1
+    Host: auth.alice.example
+    Authorization: DPoP ....
+  </pre>
+  <pre highlight=http line-highlight=3>
+    HTTP/1.1 200 OK
+    Link: &lt;https://projectron.example/#app&gt;;
+      anchor="https://auth.alice.example/bcf22534-0187-4ae4-b88f-fe0f9fa96659";
+      rel="http://www.w3.org/ns/solid/interop#registeredApplication"
+  </pre>
+  <figcaption>HEAD request to and response from Authorization Agent</figcaption>
+</figure>
+
+## User Consent ## {#user-consent}
+
+In case where application haven't been registered yet it needs to initiate flow with Authorization Agent.
+
+Issue: Add details as soon as defined for Authorization Agent
+
+After successful flow aplication will be able to discover its registration.
+
+# Application Registration # {#application-registration}
+
+Application Registration can be considered an entry point to all the data
+that user authorized it to access. Next step in discovery of that data
+is the Access Receipt linked via <code>interop:hasAccessReceipt</code> predicate.
+
```

Comment:
Will need to incorporate the adjustments from Agent Registration in #169

Reply:
Captured in issue on line 156